src/components/toast/__tests__/Toast.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { act, render, screen } from "@testing-library/react";
+import Toast from "../Toast";
+import { vi } from "vitest";
+import userEvent from "@testing-library/user-event";
+
+const toastMessage = "toast message";
+const delay = 1000;
+const mockCloseHandler = vi.fn();
+
+describe("Toast component tests", () => {
+
+  test("renders toast component when isOpen is set true", async () => {
+    render(
+      <Toast
+        isOpen={true}
+        variant={undefined}
+        severity={undefined}
+        closeHandler={mockCloseHandler}
+        message={toastMessage}
+      />
+    );
+
+    expect(screen.getByText(toastMessage)).toBeInTheDocument();
+  });
+
+  test("toast is hidden when isOpen is set false", () => {
+    render(
+      <Toast
+        isOpen={false}
+        closeHandler={mockCloseHandler}
+        severity={undefined}
+        variant={undefined}
+        message={toastMessage}
+      />
+    );
+
+    expect(screen.queryByText("Test Message")).not.toBeInTheDocument();
```

Comment:
Hmm maybe like this? We want to make sure the expected toast with the expected message is not displayed if the isOpen is false, why querying for another text?

```suggestion
    expect(screen.queryByText(toastMessage)).not.toBeInTheDocument();
```